crates/kornia-algebra/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+//! Lie groups and algebras for kornia-rs.
+//!
+//! This crate provides:
+//! - Unified algebraic types (`types` module) using newtype pattern over glam backend
+//! - Lie group implementations (SO2, SO3, SE2, SE3)
+
+// Algebraic types and Lie groups
+mod lie;
+mod mat;
+mod quat;
+mod vec;
+
+// Re-export types at crate root for convenience
+pub use lie::{SE2, SE3, SO2, SO3};
+pub use mat::{Mat2F32, Mat2F64, Mat3AF32, Mat3F32, Mat3F64, Mat4F32, Mat4F64};
+pub use quat::{QuatF32, QuatF64};
+pub use vec::{Vec2F32, Vec2F64, Vec3AF32, Vec3F32, Vec3F64, Vec4F32, Vec4F64};
+
+// Type aliases for default single-precision types.
+pub type Vec2 = Vec2F32;
```

Comment:
After reading the whole thing this is confusing for users. I would stick to one single convention and avoid maintenance burden

Reply:
Having defaults is also not good I we want to have strict typings